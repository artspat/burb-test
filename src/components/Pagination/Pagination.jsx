import React, { useMemo } from 'react';
import ReactPaginate from 'react-paginate';
import './Pagination.scss';

export const Pagination = ({ pageSize, count, onChange }) => {
    const pageCount = useMemo(() => {
        if (!pageSize || pageSize <= 0) {
            return 1;
        }
        
        return Math.ceil(count / pageSize);
    }, [pageSize, count]);

    return (
        <ReactPaginate
            nextLabel={'Next'}
            previousLabel={'Prev'}
            breakLabel={'...'}
            pageCount={pageCount}
            pageRangeDisplayed={5}
            marginPagesDisplayed={2}
            
            onPageChange={onChange}

            containerClassName="b-pagination"
            pageClassName="b-pagination__page"
            nextClassName="b-pagination__page"
            breakClassName="b-pagination__break"
            previousClassName="b-pagination__page"
            activeClassName="b-pagination__page--active"
        />
    );
};
