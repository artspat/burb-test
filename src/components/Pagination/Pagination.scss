@import '../../styles/global.scss';

.b-pagination {
    padding: 0;
    display: flex;
    list-style: none;
    flex-wrap: nowrap;
    flex-direction: row;

    &__page {
        cursor: pointer;
        padding: 3px 7px;
        margin-right: 3px;
        border-radius: 3px;
        border: 1px solid $grey;

        &:last-child {
            margin-right: 0;
        }

        &--active {
            cursor: default;
            background-color: $grey;
        }

        &.disabled {
            color: $grey;
            cursor: default;
        }

        a:focus {
            outline: none;
        }
    }

    &__break {
        padding: 3px 7px;
    }
}