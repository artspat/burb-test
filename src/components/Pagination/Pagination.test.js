import React from 'react';
import { shallow } from 'enzyme';
import { Pagination } from './Pagination';

describe('<Pagination />', () => {
    describe('when props are passed', () => {
        it('should render component', () => {
            const wrapper = shallow(<Pagination pageSize={10} count={100} />);
            expect(wrapper).toMatchSnapshot();
        });
    });

    describe('when no page size is not valid', () => {
        it('should not render component', () => {
            const wrapper = shallow(<Pagination pageSize={0} count={100} />);
            expect(wrapper).toMatchSnapshot();
        });
    });
});
