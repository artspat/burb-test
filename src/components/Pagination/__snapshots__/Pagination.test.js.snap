// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> when no page size is not valid should not render component 1`] = `
<o
  activeClassName="b-pagination__page--active"
  breakClassName="b-pagination__break"
  breakLabel="..."
  containerClassName="b-pagination"
  disableInitialCallback={false}
  disabledClassName="disabled"
  eventListener="onClick"
  marginPagesDisplayed={2}
  nextAriaLabel="Next page"
  nextClassName="b-pagination__page"
  nextLabel="Next"
  pageClassName="b-pagination__page"
  pageCount={1}
  pageRangeDisplayed={5}
  previousAriaLabel="Previous page"
  previousClassName="b-pagination__page"
  previousLabel="Prev"
/>
`;

exports[`<Pagination /> when props are passed should render component 1`] = `
<o
  activeClassName="b-pagination__page--active"
  breakClassName="b-pagination__break"
  breakLabel="..."
  containerClassName="b-pagination"
  disableInitialCallback={false}
  disabledClassName="disabled"
  eventListener="onClick"
  marginPagesDisplayed={2}
  nextAriaLabel="Next page"
  nextClassName="b-pagination__page"
  nextLabel="Next"
  pageClassName="b-pagination__page"
  pageCount={10}
  pageRangeDisplayed={5}
  previousAriaLabel="Previous page"
  previousClassName="b-pagination__page"
  previousLabel="Prev"
/>
`;
